src/debug.cpp

```diff
@@ -247,18 +247,59 @@ std::ostream &operator<<( std::ostream &out, DebugClass cl )
     return out;
 }
 
+struct time_info {
+    int hours;
+    int minutes;
+    int seconds;
+    int mseconds;
+
+    template <typename Stream>
+    friend Stream& operator<<(Stream& out, time_info const& t) {
+        using char_t = typename Stream::char_type;
+        using base   = std::basic_ostream<char_t>;
+
+        static_assert(std::is_base_of<base, Stream>::value, "");
+
+        out << t.hours   << ':' << t.minutes << ':'
+            << t.seconds << '.' << t.mseconds;
+
+        return out;
+    }
+};
+
+#ifdef _MSC_VER 
+inline time_info get_time() noexcept {
```

Comment:
Avoid inline, the compiler has more information than you do to make that decision.